code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -146,6 +146,15 @@
 	to_chat(user, "<span class='notice'>You swipe \the [src]. A console window fills the screen, but it quickly closes itself after only a few lines are written to it.</span>")
 	return FALSE
 
+/obj/item/modular_computer/microwave_act(obj/machinery/microwave/M)
+	. = ..()
+	var/obj/item/computer_hardware/battery/battery_module = all_components[MC_CELL]
+	if(battery_module && battery_module.battery && battery_module.battery.charge < battery_module.battery.maxcharge * 0.4)
+		src.give_power(battery_module.battery.maxcharge * 0.4 - battery_module.battery.charge)
```

Comment:
Cant this be negative?

Reply:
no, the prior condition checks that the charge is less than 40%